1-Draft/RFC00XX-Update-Error-View.md

```diff
@@ -0,0 +1,204 @@
+
+```powershell
+PS C:\> .\MyScript.ps1
+ERROR: ItemNotFoundException
+  ---> C:\GitHub\pri-errorview\RustTest\test.ps1
+    |
+15  | Get-ChildItem -Path c:\notreal
+    |                     ^^^ Cannot find path 'C:\notreal' because it does not exist.
+    |
+    * Help: Additional help information provided here
+```
+
+3. The 'Dynamic' view is a combination of the views 'message' and 'Analytic' for users working
+in the console that also run scripts. When the user is working Interactively in the CLI,
+they will receive errors in the view of 'Message'.  When the user executes a script,
+errors will use the view 'Analytic'.
+
+- An example is in the image below:
+
+![Message and Analytic](.\RFC00XX-Update-Error-View.png)
+
+4. A new cmdlet `Resolve-ErrorRecord` will produce comprehensive detailed
+view of the fully qualified error, including nested inner exceptions.
+
+- Resolve-ErrorRecord will provide the following:
+
+    + Display Last Error ($Error[0]) – default behavior
+    + Accept Pipeline input – support $error[1] |
+    + Option for the Newest X errors in the session
+
+- Resolve-ErrorRecord syntax
+
+```powershell
+Resolve-ErrorRecord  [-InputObject <psobject>] [-Newest <Int32>] [-All] [<CommonParameters>]
+```
+
+First parameter set
+
+- Newest
+
+    + Datatype: int32
+    + specifies one or more of the newest errors to display
+    + Not required
+
+__Example 1__
+Error occurs in Interactive mode. Cmdlet displays details of the last error displayed
+
+```powershell
+PS C:\> Get-Childitem -Path c:\notreal
+ERROR: Cannot find path ‘C:\notreal’ because it does not exist
+
+PS C:\test> Resolve-ErrorRecord
+
+**** Detailed message here ****
+```
+
+__Example 2__
+Error occurs in script, shows error from view 'Analytic', and then is piped
+from $error array to 'Resolve-ErrorRecord' to display more details.
+
+```powershell
+PS C:\> .\MyScript.ps1
+ERROR: ItemNotFoundException
+  ---> C:\GitHub\pri-errorview\RustTest\test.ps1
+    |
+15  | Get-ChildItem -Path c:\notreal
+    |                     ^^^ Cannot find path 'C:\notreal' because it does not exist.
+    |
+    * Help: this is for additional help information
+
+PS C:\> $error[0] | Resolve-ErrorRecord
+
+**** Detailed message here ****
+```
+
+__Example 3__
+Display detailed error information for the most recent 3 errors.
+
+```powershell
+PS C:\> Resolve-ErrorRecord -Newest 3
+
+**** Detailed message here ****
+**** Detailed message here ****
+**** Detailed message here ****
+```
+
+## Alternate Proposals and Considerations
+
+__Alternative single-line display__
+
+```powershell
+<CommandName>:<Exception Message>
+get-item: Cannot find path ‘C:\blah’ because it does not exist:
+
+<CommandName>:<Exception Message>: <Position>
+get-item: Cannot find path ‘C:\blah’ because it does not exist: At line:1 char:1
+
+<CommandName>:<Alias>:<Exception Message>: <Position>
+get-item: gi: Cannot find path ‘C:\blah’ because it does not exist: At line:1 char:1
+
```

Comment:
I would find this very confusing if I did not use the alias.

Similarly, I wouldn't expect <CommandName> to be expanded. If I used "icm", I want the script line error to report that, not Invoke-Command!

Reply:
Currently, the error message produced displays the full cmdlet name, even if you specified an alias.  The reason is for clarity in the error message.

Reply:
I think adding the alias does make it confusing and isn't in the current code.

Reply:
Better to show what was invoked, and nothing more than that. If you invoke an alias, show the alias. If you invoke a cmdlet or function or script, show the command or function or script.